hail/src/main/scala/is/hail/backend/service/ServiceBackend.scala

```diff
@@ -715,6 +744,63 @@ class ServiceBackendSocketAPI2(
             writeString(formatException(t))
         }
 
+      case INDEX_BGEN =>
+        val tmpdir = readString()
+        val billingProject = readString()
+        val remoteTmpDir = readString()
+        val nFiles = readInt()
+        val files = new Array[String](nFiles)
+        var i = 0
+        while (i < nFiles) {
+          files(i) = readString()
+          i += 1
+        }
+        val nIndexFiles = readInt()
+        val indexFileMap = mutable.Map[String, String]()
+        i = 0
+        while (i < nIndexFiles) {
+          val k = readString()
+          val v = readString()
+          indexFileMap(k) = v
+          i += 1
+        }
+        val hasReferenceGenome = readBool()
+        val referenceGenomeName = hasReferenceGenome match {
+          case true => Some(readString())
+          case false => None
+        }
+        val nContigRecoding = readInt()
+        val contigRecoding = mutable.Map[String, String]()
+        i = 0
+        while (i < nContigRecoding) {
+          val k = readString()
+          val v = readString()
+          contigRecoding(k) = v
+          i += 1
+        }
+        val skipInvalidLoci = readBool()
+
+        try {
+          val result = backend.indexBgen(
```

Comment:
`result` in unused so might as well not assign

Reply:
I made indexBgen return the string instead for symmetry with the other methods.